sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -688,6 +688,7 @@ pi_result _pi_device::initialize(int SubSubDeviceOrdinal,
   if (SubSubDeviceOrdinal >= 0) {
     QueueGroup[queue_group_info_t::Compute].ZeOrdinal = SubSubDeviceOrdinal;
     QueueGroup[queue_group_info_t::Compute].ZeIndex = SubSubDeviceIndex;
+    QueueGroup[queue_group_info_t::Compute].ZeProperties.numQueues = 1;
```

Comment:
Should we initialize all fields of ZeProperties (there are more of them, see https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=numqueues#ze-command-queue-group-properties-t)?

Can we reuse  zeDeviceGetCommandQueueGroupProperties result from here https://github.com/intel/llvm/pull/6349/files#diff-15dd1eb076d2164bd9e87d9057f05f652a716498e8cdf5975e564c65309a0985R684-R685? If I am not mistaken ZeProperties is supposed to be initialized using zeDeviceGetCommandQueueGroupProperties and not manually.

Reply:
L0 backend does not have any subsub devices. In the current implementation, we piggy back on the sub-device to act as the ZeDevice for subsubdevice. So, the zeDeviceGetCommandQueueGroupProperties  returns the properties for sub-device and not the subsub device. So, that's why I am not using it here.

Reply:
Also, about initializing all other fields. I think we can try to initialize minimal number of fields. Currently, we need only the number of queues.

Reply:
Sorry, but I have troubles understanding why it is ok to initialize only numQueues.

For example, at this part of code in enqueueMemFillHelper we use maxMemoryFillPatternSize:
https://github.com/intel/llvm/blob/sycl/sycl/plugins/level_zero/pi_level_zero.cpp#L6399-L6419

If Device at this part of code is a sub-sub device then we use uninitialized field maxMemoryFillPatternSize there.

Reply:
As @asudarsa mentions, L0 doesn't have sub-sub-devices, to the properties you can get are the ones from the sub-device. I think that we have two options:

- whenever querying for props of a sub-sub-devices, query the ones from the parent sub-devices, either sub-device->props or a pointer to it.
- copy them at initialization.

Reply:
I agree for sub-sub-device we need to initialize the entire `ZeProperties` (of type `ze_command_queue_group_properties_t`), populate it with what it's parent sub-device has in `zeDeviceGetCommandQueueGroupProperties` and then update as makes sense, e.g. set `numQueues = 1`

Reply:
Hi @smaslov-intel and @jandres742 

Thanks so much for the feedback. As you mentioned, I think it is a good idea to initialize the entire ZeProperties structure using data gathered for sub-device. However, zeDeviceGetCommandQueueGroupProperties returns multiple ZeProperties, one for each group. I think I will try to use the one for 'compute' group. Please let me know if there are questions.

Thanks again

Reply:
Added a new commit addressing changes. Thanks

Reply:
Looks good, but please add a comment why "numQueues" is updated for sub-sub-device